cmd/traefik/traefik.go

```diff
@@ -360,7 +360,23 @@ func getHTTPChallengeHandler(acmeProviders []*acme.Provider, httpChallengeProvid
 
 func getDefaultsEntrypoints(staticConfiguration *static.Configuration) []string {
 	var defaultEntryPoints []string
+
+	// Determines if at least one EntryPoint is configured to be in the default set.
+	var userDefinedDefaults bool
+	for _, ep := range staticConfiguration.EntryPoints {
+		if ep.DefaultSet {
+			userDefinedDefaults = true
+			break
+		}
+	}
+
 	for name, cfg := range staticConfiguration.EntryPoints {
+		// Ignores EntryPoints that are marked as not part of default set,
+		// if at least one is configured to be in the default set.
```

Comment:
```suggestion
		// By default, if no entrypoints exist on the default set, all entrypoints are considered.
		// If at least one is included on the default set, then all entrypoints are ignored unless flagged.
```
not sure on the actual wording here but I'm trying to make it clearer, as the original comment wasn't for me :)

Reply:
What about?

```suggestion
		// By default all entrypoints are considered.
		// If at least one is flagged, then only flagged entrypoints are included.
```